### PR TITLE
Keep/Merge/Remove decision rubric

### LINKED ISSUE #385: Keep/Merge/Remove decision rubric
## Summary
Define a clear rubric to classify features into keep, merge, or remove categories based on objective criteria.

## Scope
- Specify threshold rules tied to value and maintenance scoring.
- Add tie-break rules for safety, migration risk, and user disruption.
- Document how rubric decisions map to release phases in v0.14.x.

## Acceptance criteria
- The rubric yields deterministic outcomes for sample feature candidates.
- The team can apply the rubric consistently without ad-hoc judgment.

### PR DESCRIPTION
**Pull request by utilForever**
## What

Add a deterministic keep/merge/remove rubric to feature inventory reporting, including explicit threshold criteria, boundary tie-break rules, release phase mapping for v0.14.x, and regenerated report artifacts.

## Why

Issue #385 asked for a clear and repeatable decision rubric so recommendations are not ad-hoc and are mapped to actionable release phases. This update makes the scoring contract explicit in code and in generated reports.

Closes #385.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #25`)

### Functional Validation

- [ ] Pomodoro timer behavior was verified for this change (if applicable)
- [ ] Site blocking behavior was verified for this change (if applicable)
- [ ] WakaTime integration behavior was verified for this change (if applicable)
- [x] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [ ] User-facing docs were updated (`README.md` or relevant docs, if applicable)
- [ ] New dependencies/configuration are documented (if applicable)
- [x] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed)
- [ ] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **New Features**
  * Enhanced scoring with tie‑break logic for edge cases and release‑phase mapping aligning recommendations to v0.14.x guidance.

* **Documentation**
  * Updated feature inventory docs describing tie‑break rules and release phase assignments.

* **Tests**
  * Added/updated tests validating tie‑break behavior, release‑phase coverage, and deterministic rubric/report outputs.

* **Chores**
  * Bumped feature inventory schema version and regenerated/ordered CLI flag coverage for deterministic summaries.

### CODE REVIEW

File: src/feature_inventory.rs [hunk 1]
```diff
@@ -560,7 +592,20 @@ pub fn render_markdown_report(report: &FeatureInventoryReport) -> String {
...
+    ));
+    markdown.push_str(&format!(
+        "- Keep tie-break (delta == {:.2}): keep when any of safety/migration_risk/user_disruption >= {}\n",
+        KEEP_MIN_DELTA, TIE_BREAK_KEEP_SIGNAL_MIN
+    ));
+    markdown.push_str(&format!(
+        "- Remove tie-break (delta == {:.2}): remove when safety/migration_risk/user_disruption are all <= {}\n",
+        REMOVE_MAX_DELTA, TIE_BREAK_REMOVE_SIGNAL_MAX
+    ));
+    markdown.push_str("- Tie-break dimensions: safety = failure_impact, migration_risk = complexity, user_disruption = support_burden\n\n");
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/feature_inventory.rs:610]: Good catch. Updated 
ender_markdown_report to read keep/remove thresholds and tie-break values from 
eport.scoring_model/	ie_break_model, so markdown and JSON stay aligned with the same report model. Fixed in b7eba29.
